File: starter_kits/Java/hlt/EntityId.java
package hlt;

public class EntityId {
    public static final EntityId NONE = new EntityId(-1);

    public final int id;

    public EntityId(int id) {
        this.id = id;
    }

    @Override
    public String toString() {
        return String.valueOf(id);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        EntityId entityId = (EntityId) o;

        return id == entityId.id;
    }

    @Override
    public int hashCode() {
        return id;
    }
}

File: starter_kits/PHP/hlt/EntityId.php
<?php

namespace hlt;

class EntityId {
    /**
     * @var int
     */
    public $id;

    public function __construct(int $id = -1)
    {
        $this->id = $id;
    }

    public function __toString()
    {
        return '' . $this->id;
    }

    public function equals($other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }

        return $this->id === $other->id;
    }
}
